### PR TITLE
Devops 377: add `reallocate-core-ips` subcommand to `iohk-ops`

### PR DESCRIPTION
**Pull request by deepfire**

### CODE REVIEW

File: iohk/NixOps.hs [hunk 1]
```diff
@@ -408,7 +408,7 @@ readTopology file = do
     Left err -> errorT $ format ("Failed to parse topology file: "%fp%": "%w) file err
 
 data SimpleTopo
-  =  SimpleTopo (Map.Map NodeName SimpleNode)
+  =  SimpleTopo { fromSimpleTopo :: (Map.Map NodeName SimpleNode) }
```
Comment (domenkozar) [inline iohk/NixOps.hs]: Shouldn't this be newtype?

File: iohk/NixOps.hs [hunk 2]
```diff
@@ -445,8 +445,11 @@ instance ToJSON NodeType
 topoNodes :: SimpleTopo -> [NodeName]
 topoNodes (SimpleTopo cmap) = Map.keys cmap
 
+topoCores :: SimpleTopo -> [NodeName]
+topoCores = (fst <$>) . filter ((== NodeCore) . snType . snd) . Map.toList . fromSimpleTopo
+
 topoNCores :: SimpleTopo -> Int
-topoNCores (SimpleTopo cmap) = Map.size $ flip Map.filter cmap ((== NodeCore) . snType)
+topoNCores = length . topoCores
```
Comment (domenkozar) [inline iohk/NixOps.hs]: This function seems redundant now

File: iohk/iohk-ops.hs [hunk 3]
```diff
@@ -181,6 +182,8 @@ centralCommandParser =
    , ("destroy",                "Destroy the whole cluster",                                        pure Destroy)
    , ("delete",                 "Unregistr the cluster from NixOps",                                pure Delete)
    , ("fromscratch",            "Destroy, Delete, Create, Deploy",                                  pure FromScratch)
+   , ("reallocate-core-ips",    "Destroy elastic IPs corresponding to the nodes listed and reprovision cluster",
```
Comment (domenkozar) [inline iohk/iohk-ops.hs]: Reprovision means that resources would be reprovisioned, we should just say "deploy"

File: iohk/NixOps.hs (hunk 1 above)
Comment (deepfire) [reply iohk/NixOps.hs]: It started as a `newtype`, right -- but for some reason I had to turn this into a `data`.  Don't remember why : -/

Besides, there is no _real_ difference (except performance, which I think we can agree is not an issue here) between these options.

File: iohk/NixOps.hs (hunk 2 above)
Comment (deepfire) [reply iohk/NixOps.hs]: You seem to refer to the fact that `topoNCores` should now be inlined (in two locations, as of now)?

I find it arguable, but OK.  Done.

File: iohk/iohk-ops.hs (hunk 3 above)
Comment (deepfire) [reply iohk/iohk-ops.hs]: Done.

**Comment by domenkozar** [comment]
Lastly, this needs a section in operational manual. What is the symptom and what to run as a solution (this new command).

**Comment by deepfire** [comment]
@domenkozar, what do you refer to by symptoms -- does that mean assessment of whether or not we are under a DDoS?  That would probably include DataDog metrics (some of which we might already have?).

If this is what you mean, do we already have a DDoS response section in the manual?

Again, if this is what you mean, I think we might need to extend the scope of https://issues.serokell.io/issue/DEVOPS-377 to cover documentation of DDoS response -- right now it's clearly not in the task description.

**Comment by domenkozar** [comment]
Ah, this wasn't made against staging. I'll reopen.

**Pull request merged**